File: examples_ios/MyApp/Sources/Controllers/ViewModels/TriggerOverviewViewModel.swift
import ConfirmitMobileSDK
import Foundation

class TriggerOverviewViewModel {
  private(set) var program: ProgramModel!

  init(program: Program) throws {
    self.program = try ProgramModel(program: program)
  }

  func notifyEvent(event: String, data: [String: String]) {
    TriggerSDK.notifyEvent(serverId: program.program.serverId, programKey: program.program.programKey, event: event, data: data)
  }

  func removeProgram() -> Bool {
    do {
      try TriggerSDK.deleteProgram(serverId: program.program.serverId, programKey: program.program.programKey, deleteCustomData: true)
      TriggerSDK.removeCallback(serverId: program.program.serverId, programKey: program.program.programKey)
      return true
    } catch {
      return false
    }
  }

  func downloadCounters() {
    TriggerSDK.downloadCounters(serverId: program.program.serverId, programKey: program.program.programKey) { _, _ in }
  }

  func downloadProgram(completion: @escaping () -> Void) {
    TriggerSDK.download(serverId: program.program.serverId, programKey: program.program.programKey) { _, _ in
      completion()
    }
  }
}

File: examples_android/app/src/main/java/com/example/myapp/fragments/viewmodels/TriggerOverviewViewModel.kt
package com.example.myapp.fragments.viewmodels

import androidx.lifecycle.ViewModel
import com.confirmit.mobilesdk.TriggerSDK
import com.confirmit.mobilesdk.database.externals.Program
import com.example.myapp.fragments.viewmodels.models.ProgramModel

class TriggerOverviewViewModel : ViewModel() {
    lateinit var program: ProgramModel
        private set

    fun setProgramModel(program: Program) {
        this.program = ProgramModel(program)
    }

    suspend fun downloadCounters() {
        TriggerSDK.downloadCountersAsync(program.program.serverId, program.program.programKey).await()
    }

    fun notifyEvent(event: String, data: Map<String, String?>) {
        TriggerSDK.notifyEvent(program.program.serverId, program.program.programKey, event, data)
    }

    suspend fun downloadProgram() {
        TriggerSDK.downloadAsync(program.program.serverId, program.program.programKey).await()
    }

    fun removeProgram(): Boolean {
        return try {
            TriggerSDK.deleteProgram(program.program.serverId, program.program.programKey, true)
            TriggerSDK.removeCallback(program.program.serverId, program.program.programKey)
            true
        } catch (_: Exception) {
            false
        }
    }
}